framework/include/transfers/MultiAppNearestNodeTransfer.h

```diff
@@ -96,4 +96,12 @@ class MultiAppNearestNodeTransfer : public MultiAppConservativeTransfer
   std::map<processor_id_type, std::vector<dof_id_type>> & _cached_dof_ids;
   std::map<dof_id_type, unsigned int> & _cached_from_inds;
   std::map<dof_id_type, unsigned int> & _cached_qp_inds;
+
+private:
+  /**
+   * Get the local nodes on the target boundary for the transfer
+   * @param to_problem_id index of the problem transfer is sending to
+   * @return target local nodes
+   */
+  std::vector<Node *> getTargetLocalNodes(const unsigned int to_problem_id);
```

Comment:
It might be a good idea to avoid copying if `std::vector` is large

Reply:
You mean as in, use a reference instead?

If so, I was under the impression that because we're creating `target_local_nodes` like

```
std::vector<Node *> target_local_nodes;

target_local_nodes.push_back(stuff);
```

that you can't return a reference to `target_local_nodes`, since it's scope is only within the `getTargetLocalNodes` function. 

Or did you mean not using `std::vector<Node *>`, but `std::vector<Node>` instead?

Reply:
I think we would need to create the vector before calling the routine to use a reference.